internal/flows/init.go

```diff
@@ -28,6 +28,16 @@ func (i *Initer) Run(ctx context.Context) error {
 		return err
 	}
 
+	// in the case of using iam roles anywhere, with EnableCredentialsFile
+	// we want to start the service before configuring aws so we can use the
+	// credentials file to configure aws
+	if !slices.Contains(i.SkipPhases, preprocessPhase) {
+		i.Logger.Info("Configuring Pre-process daemons...")
+		if err := i.NodeProvider.PreProcessDaemon(ctx); err != nil {
+			return err
+		}
+	}
```

Comment:
instead of moving this block up, what if we move the "create update service" to the `ConfigureAws` method? i think that's probably where it belong now, in the same way that `ConfigureAws` does the ssm registration to make sure the node has credentials.